commcare_export/minilinq.py

```diff
@@ -343,7 +343,10 @@ def eval(self, env):
         fn_result = self.fn.eval(env)
         args_results = [arg.eval(env) for arg in self.args]
 
-        return fn_result(*args_results)
+        result = fn_result(*args_results)
+        if isinstance(result, MiniLinq):
+            return result.eval(env)
```

Comment:
This allows functions to return minilinq which gets recursively evaluated instead of just static values.